client/boot/index.js

```diff
@@ -165,14 +166,7 @@ function boot() {
 function renderLayout( reduxStore ) {
 	const Layout = require( 'controller' ).ReduxWrappedLayout;
 
-	const layoutElement = React.createElement( Layout, {
-		store: reduxStore
-	} );
-
-	ReactDom.render(
-		layoutElement,
-		document.getElementById( 'wpcom' )
-	);
+	renderWithReduxStore( React.createElement( Layout, { store: reduxStore } ), 'wpcom', reduxStore );
```

Comment:
I might be wrong but I think we were not using the JSX syntax in `.js` files (only`.jsx`) for historic reason which are not valid anymore.